datafusion/optimizer/src/eliminate_duplicated_expr.rs

```diff
@@ -35,78 +35,107 @@ impl EliminateDuplicatedExpr {
         Self {}
     }
 }
-
+// use this structure to avoid initial clone
```

Comment:
```suggestion
/// Wrap the Expr in a Wrapper to support specialized comparison.
///
/// Ignores the sort options for `SortExpr` because if the expression is the same
/// the subsequent exprs are never matched
/// 
/// For example,  `ORDER BY a ASC a DESC` is the same
// as `ORDER BY a ASC` (the second `a DESC` is never compared)
```

Reply:
Sure!